File: cocos2d-x-2.2.0/projects/Ironman/Classes/GameSceneSetLayer.cpp
//
//  GameSceneSetLayer.cpp
//  Ironman
//
//  Created by Ken on 13-10-17.
//
//

#include "GameSceneSetLayer.h"
#include "AudioPlayer.h"

bool GameSceneSetLayer::init(int effectStatus, int volumn)
{
	if(UILayer::init()){
		
		GameScene* parentScene = GameScene::shareGameScene();
		parentScene->gameSceneMapLayer->stop();
		parentScene->stopAllActions();
		parentScene->playLayer->stopAllActions();
		parentScene->playLayer->setTouchEnabled(false);
		parentScene->gameSceneMonster->MonsterAmature->pauseSchedulerAndActions();
		parentScene->playLayer->imManArmature->pauseSchedulerAndActions();
		parentScene->menuLayer->settingBtn->setTouchEnable(false);
		
		this->addWidget(dynamic_cast<Layout*>(CCUIHELPER->createWidgetFromJsonFile("GameSceneSetMenu_1.json")));
		
		musicEffectSlider = dynamic_cast<UISlider*>(this->getWidgetByName("musicEffect"));
		musicVolumeSlider = dynamic_cast<UISlider*>(this->getWidgetByName("musicVolume"));
		UIButton* backGameBtn = dynamic_cast<UIButton*>(this->getWidgetByName("backGame"));
		UIButton* returnMainMenuBtn = dynamic_cast<UIButton*>(this->getWidgetByName("returnMainMenu"));
		
		musicEffectStatus = effectStatus;
		
		if(musicEffectStatus == 0){
			
			musicEffectSlider->setPercent(15);
		}else if(musicEffectStatus == 1){
			
			musicEffectSlider->setPercent(95);
		}
		musicVolumeSlider->setPercent(volumn);
		
		backGameBtn->addTouchEventListener(this, toucheventselector(GameSceneSetLayer::backGameBtn));
		returnMainMenuBtn->addTouchEventListener(this, toucheventselector(GameSceneSetLayer::returnMainMenuBtnFunc));
		musicEffectSlider->addEventListener(this, sliderpercentchangedselector(GameSceneSetLayer::musicEffectSliderCallFunc));
		musicVolumeSlider->addEventListener(this, sliderpercentchangedselector(GameSceneSetLayer::musicVolumeSliderCallFunc));
		
		return true;
	}
	
	return false;
}

void GameSceneSetLayer::musicEffectSliderCallFunc(cocos2d::CCObject *pSender, SliderEventType type)
{
	if(type == SLIDER_PERCENTCHANGED){
		
		if(musicEffectStatus == 0){
			
			musicEffectSlider->setPercent(95);
			musicEffectStatus=1;
		}else if(musicEffectStatus == 1){
			
			musicEffectSlider->setPercent(15);
			musicEffectStatus=0;
		}
	}
	
	GameScene::shareGameScene()->menuLayer->musicEffect = musicEffectStatus;
    //set effect state.
    CCLog("musicEffectStatus:%d.", musicEffectStatus);
    if (musicEffectStatus == 1) {
        AudioPlayer::sharedAudio()->setEffectPlay(false);
    }
    else{
        AudioPlayer::sharedAudio()->setEffectPlay(true);
    }
}

void GameSceneSetLayer::musicVolumeSliderCallFunc(cocos2d::CCObject *pSender, SliderEventType type)
{
    float voice = 0.0f;
	if(type == SLIDER_PERCENTCHANGED){
		voice = musicVolumeSlider->getPercent();
		if(musicVolumeSlider->getPercent()<8){
			musicVolumeSlider->setPercent(8);
            voice=0.0f;
		}else if(musicVolumeSlider->getPercent()>95){
			musicVolumeSlider->setPercent(95);
            voice=100.0f;
		}
	}
	
	GameScene::shareGameScene()->menuLayer->musicVolume = musicVolumeSlider->getPercent();
    //set audio voice.
    AudioPlayer::sharedAudio()->setVolume(voice/100);
}

void GameSceneSetLayer::backGameBtn(cocos2d::CCObject *pSender, TouchEventType type)
{
	if(TOUCH_EVENT_ENDED == type){
		
		GameScene* parentScene = GameScene::shareGameScene();
		parentScene->resumeSchedulerAndActions();
	    parentScene->playLayer->imManArmature->resumeSchedulerAndActions();
		parentScene->playLayer->setTouchEnabled(true);
		
		std::string currentMovementId = parentScene->playLayer->imManArmature->getAnimation()->getCurrentMovementID();
		CCLog("currentMovementId is %s", currentMovementId.c_str());
		if(currentMovementId.compare("") !=0 && (currentMovementId.compare("Running")==0 || currentMovementId.compare("RuningJump")==0))
			parentScene->gameSceneMapLayer->move();
		
		
		parentScene->gameSceneMonster->MonsterAmature->resumeSchedulerAndActions();
		parentScene->menuLayer->settingBtn->setTouchEnable(true);
		
	    this->removeAllChildren();
	    this->removeFromParentAndCleanup(true);
	}
}

void GameSceneSetLayer::returnMainMenuBtnFunc(cocos2d::CCObject *pSender, TouchEventType type)
{
	MainMenuScene* mainMenuScene =  new MainMenuScene();
	mainMenuScene->init();
	
	CCTransitionFade* mainMenuSceneTransition =  CCTransitionFade::create(0.5, mainMenuScene, ccWHITE);
	CCDirector::sharedDirector()->replaceScene(mainMenuSceneTransition);
}

File: cocos2d-html5_i/IronCity/src/layer/GameSceneSetLayer.js
/**
 * Created with JetBrains WebStorm.
 * User: cocos
 * Date: 13-10-23
 * Time: 下午2:59
 * To change this template use File | Settings | File Templates.
 */

var GameSceneSetLayer = cc.UILayer.extend({
    musicEffectStatus:0,
    musicEffectSlider:null,
    musicVolumeSlider:null,
    parentScene:null,
    init:function(effectStatus, volumn){
        if( cc.UILayer.prototype.init.call(this) ){
            this.parentScene = GameScene.getScene();
            this.parentScene.moveMap.stop();
            this.parentScene.stopAllActions();
            this.parentScene.playLayer.stopAllActions();
            this.parentScene.playLayer.setTouchEnabled(false);
            //parentScene.gameSceneMonster.MonsterAmature.pauseSchedulerAndActions();
            this.parentScene.playLayer.imManArmature.pauseSchedulerAndActions();
            this.parentScene.menuLayer.settingBtn.setTouchEnable(false);

            this.addWidget(cc.UIHelper.getInstance().createWidgetFromJsonFile(Json_GameSceneSetMenu_1));

            this.musicEffectSlider = this.getWidgetByName("musicEffect");
            this.musicVolumeSlider = this.getWidgetByName("musicVolume");
            var backGameBtn = this.getWidgetByName("backGame");
            var returnMainMenuBtn = this.getWidgetByName("returnMainMenu");

            this.musicEffectStatus = effectStatus;

            if(this.musicEffectStatus == 0){

                this.musicEffectSlider.setPercent(15);
            }else if(this.musicEffectStatus == 1){

                this.musicEffectSlider.setPercent(95);
            }
            this.musicVolumeSlider.setPercent(volumn);

            backGameBtn.addTouchEventListener(this, this.backGameBtn);
            returnMainMenuBtn.addTouchEventListener(this, this.returnMainMenuBtnFunc);
            this.musicEffectSlider.addEventListener(this, this.musicEffectSliderCallFunc);
            this.musicVolumeSlider.addEventListener(this, this.musicVolumeSliderCallFunc);
            return true;
        }

        return false;
    },
    musicEffectSliderCallFunc:function(pSender, type){
        if(type == cc.SliderEventType.PERCENTCHANGED){
            if(this.musicEffectStatus == 0){
                this.musicEffectSlider.setPercent(95);
                this.musicEffectStatus=1;
            }else if(this.musicEffectStatus == 1){

                this.musicEffectSlider.setPercent(15);
                this.musicEffectStatus=0;
            }
        }

        this.parentScene.menuLayer.musicEffect = this.musicEffectStatus;
        //set audio state.
        //AudioPlayer::sharedAudio()->setBackgroundMusicPlay(musicEffectStatus);
    },
    musicVolumeSliderCallFunc:function(pSender, type){
        var voice = 0.0;
        if(type == cc.SliderEventType.PERCENTCHANGED){
            voice = this.musicVolumeSlider.getPercent();
            if(this.musicVolumeSlider.getPercent()<8){
                this.musicVolumeSlider.setPercent(8);
                voice=0.0;
            }else if(this.musicVolumeSlider.getPercent()>95){
                this.musicVolumeSlider.setPercent(95);
                voice=100.0;
            }
        }

        this.parentScene.menuLayer.musicVolume = this.musicVolumeSlider.getPercent();
        //set audio voice.
        //AudioPlayer::sharedAudio().setVolume(voice/100);
    },
    backGameBtn:function(pSender, type){
        if(cc.TouchEventType.ENDED == type){
            this.parentScene.resumeSchedulerAndActions();
            this.parentScene.playLayer.imManArmature.resumeSchedulerAndActions();
            this.parentScene.playLayer.setTouchEnabled(true);

            var currentMovementId = this.parentScene.playLayer.imManArmature.getAnimation().getCurrentMovementID();
            cc.log("currentMovementId is %s", currentMovementId);
            if(currentMovementId.compare("") !=0 && (currentMovementId.compare("Running")==0 || currentMovementId.compare("RuningJump")==0))
                this.parentScene.moveMap.move();

            //this.parentScene.gameSceneMonster.MonsterAmature.resumeSchedulerAndActions();
            this.parentScene.menuLayer.settingBtn.setTouchEnable(true);

            //this.removeAllChildren();
            this.removeFromParent(true);
        }
    },
    returnMainMenuBtnFunc:function(pSender, type){
        var mainMenuScene =  new MainMenuScene();
        mainMenuScene.init();

        var mainMenuSceneTransition =  cc.TransitionFade.create(0.5, mainMenuScene, cc.WHITE);
        cc.Director.getInstance().replaceScene(mainMenuSceneTransition);
    }
});